packages/shared/src/porter.ts

```diff
@@ -31,20 +37,37 @@ export const domains: Record<string, Domain> = {
   MAINNET: 'mainnet',
 };
 
-export const getPorterUri = (domain: Domain): string => {
-  return getPorterUris(domain)[0];
+export const getPorterUri = async (domain: Domain): Promise<string> => {
+  return (await getPorterUris(domain))[0];
 };
 
-export const getPorterUris = (domain: Domain, porterUris: string[] = []): string[] => {
+export const getPorterUris = async (domain: Domain, porterUris: string[] = []): Promise<string[]> => {
   const fullList = [...porterUris];
   const uri = defaultPorterUri[domain];
   if (!uri) {
     throw new Error(`No default Porter URI found for domain: ${domain}`);
   }
   fullList.push(uri);
+  const urisFromSource = await getPorterUrisFromSource(domain);
+  fullList.push(...urisFromSource);
   return fullList;
 };
 
+export const getPorterUrisFromSource = async (domain: Domain): Promise<string[]> => {
+  const source = porterUriSource[domain];
+  if (!source) {
+    return [];
+  }
+  try {
+    const resp = await axios.get(source, {
+      responseType: 'blob',
+    });
+    return resp.data.split(/\r?\n/).filter(Boolean);
```

Comment:
IMHO using a JSON file here is easier to access and deserialize (and we can even add types for the resulting data structure).

Reply:
✔️